src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/QuestionManager.java

```diff
@@ -619,4 +629,53 @@ public ChoiceDTO convertJsonAnswerToChoice(final String jsonAnswer) throws Error
     }
     return answerFromClientDTO;
   }
+
+  public GameFilter createGameFilterForRandomQuestions(final RegisteredUserDTO currentUser, final String subjects) {
+    GameFilter gameFilter = new GameFilter();
+
+    UserPreference filterQuestionsPreference = null;
+    try {
+      filterQuestionsPreference = this.userPreferenceManager.getUserPreference(
+          "DISPLAY_SETTING",
+          "HIDE_NON_AUDIENCE_CONTENT",
+          currentUser.getId());
+    } catch (SegueDatabaseException e) {
+      SegueErrorResponse error = new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR,
```

Comment:
You've created a new error response, but you're not really using it (throwing it).

Reply:
No, I had this throwing previously, and amended based on your previous comment. I was thinking, because it's a database error that will have the same functionality as returning 0 questions to the user, we just want to log when this happens, rather than blowing up the UI for the user (as per your previous feedback). Happy to do that if that's what you want, I'll have a chat with @hannah-whelan  on how she can handle this if you do.

Reply:
Yes I definitely don't want to receive a 500 error if the user hasn't defined a preference here :D  (I think that's the scenario we are talking about). It probably doesn't need an error response at all does it? Since if the preference does not exist or can't be fetched, it won't affect the filtering of questions. It depends if we want to log that anywhere in case of an actual problem fetching preferences from DB, as opposed to the user just not having one set.

Reply:
If you don't want to return a 500 error in this scenario, then that's totally fine, but in that case, as Hannah mentioned, you don't need to create an error response at all.

Reply:
Are you asking for me not to even log when there is an error with the database? Is this being monitored somewhere else? My thinking was that if we suspect issues, this log would help us investigate those issues, because this code will report any issues. 👀

Reply:
Logging is fine. Creating a `SegueErrorResponse` that you don't really use is not.